core/types/calls.cc

```diff
@@ -2067,9 +2068,12 @@ class T_proc_params : public IntrinsicMethod {
         // Going to overwrite the return type later in MetaType::dispatchCall
         targs.emplace_back(core::Types::todo());
 
-        for (size_t i = 1; i < args.args.size(); i += 2) {
-            auto unwrappedType = Types::unwrapType(gs, args.argLoc(i), args.args[i]->type);
+        core::KwPairSpan<const TypeAndOrigins *> kwspan{absl::MakeSpan(args.args)};
+        size_t i = 1;
+        for (auto [_key, value] : kwspan) {
+            auto unwrappedType = Types::unwrapType(gs, args.argLoc(i), value->type);
```

Comment:
Could we make the span over `args.args` + `locs.args`, so that we can get rid of `i`?

This is a minor nit, so feel free to ignore if it's not trivial or makes the code more complicated.

Reply:
I agree that this would be nicer if we didn't have to deal with `i`.  What we'd need in that case is some sort of `zip` iterator over multiple spans...which I think falls into the territory of a separate PR.

I would like to write up a `zip`-style iterator for iterating over kwarg pairs when the keys and values live in separate vectors; ideally once that's written, it will be general enough that it can be adapted here.